src/GafferScene/RendererAlgo.cpp

```diff
@@ -1034,120 +1034,88 @@ void outputCameras( const ScenePlug *scene, const IECore::CompoundObject *global
+		if( !camera->hasOverscanLeft() && overscanLeftData )
 		{
-			maxOffset.x = overscanValueData->readable();
+			camera->setOverscanLeft( overscanLeftData->readable() );
 		}
-		if( const FloatData *overscanValueData = globals->member<FloatData>( "option:render:overscanBottom" ) )
+		const FloatData *overscanRightData = globals->member<FloatData>( "option:render:overscanRight" );
+		if( !camera->hasOverscanRight() && overscanRightData )
 		{
-			minOffset.y = overscanValueData->readable();
+			camera->setOverscanRight( overscanRightData->readable() );
 		}
-		if( const FloatData *overscanValueData = globals->member<FloatData>( "option:render:overscanTop" ) )
+		const FloatData *overscanTopData = globals->member<FloatData>( "option:render:overscanTop" );
+		if( !camera->hasOverscanTop() && overscanTopData )
 		{
-			maxOffset.y = overscanValueData->readable();
+			camera->setOverscanTop( overscanTopData->readable() );
+		}
+		const FloatData *overscanBottomData = globals->member<FloatData>( "option:render:overscanBottom" );
+		if( !camera->hasOverscanBottom() && overscanBottomData )
+		{
+			camera->setOverscanBottom( overscanBottomData->readable() );
 		}
-
-		// convert those offsets into pixel values and apply them to the render region
-		renderRegion.min -= V2i(
-			int(minOffset.x * (float)resolution.x),
-			int(minOffset.y * (float)resolution.y)
-		);
-
-		renderRegion.max += V2i(
-			int(maxOffset.x * (float)resolution.x),
-			int(maxOffset.y * (float)resolution.y)
-		);
 	}
 
-
 	const Box2fData *cropWindowData = globals->member<Box2fData>( "option:render:cropWindow" );
-	if( cropWindowData )
+	if( !camera->hasCropWindow() && cropWindowData )
 	{
-		const Box2f &cropWindow = cropWindowData->readable();
-		Box2i cropRegion(
-			V2i( (int)( round( resolution.x * cropWindow.min.x ) ),
-			     (int)( round( resolution.y * cropWindow.min.y ) ) ),
-			V2i( (int)( round( resolution.x * cropWindow.max.x ) ) - 1,
-			     (int)( round( resolution.y * cropWindow.max.y ) ) - 1 ) );
-
-		renderRegion.min.x = std::max( renderRegion.min.x, cropRegion.min.x );
-		renderRegion.max.x = std::min( renderRegion.max.x, cropRegion.max.x );
-		renderRegion.min.y = std::max( renderRegion.min.y, cropRegion.min.y );
-		renderRegion.max.y = std::min( renderRegion.max.y, cropRegion.max.y );
-
+		camera->setCropWindow( cropWindowData->readable() );
 	}
 
-	camera->parameters()["renderRegion"] = new Box2iData( renderRegion );
-
-	// apply the shutter
-
-	camera->parameters()["shutter"] = new V2fData( SceneAlgo::shutter( globals ) );
+	// Bake the shutter from the globals into the camera before passing it to the renderer backend
+	//
+	// Before this bake, the shutter is an optional render setting override, with the shutter start
+	// and end relative to the current frame.  After baking, the shutter is currently an absolute
+	// shutter, with the frame added on.  Renderers like Arnold ask for the shutter to be relative,
```

Comment:
This is bound up with the question of `time` in OSL though isn't it, so I don't think it's quite as cut-and-dried as that?

Reply:
Perhaps weakens the argument slightly, though one could certainly imagine a renderer that correctly set up time based on a relative shutter and the current frame ( though Arnold takes AA_seed instead of a frame number ).  Regardless, I still think that this should be handled in the renderer backend if necessary, and the camera shutter should always be relative ( rather than sometimes relative and sometimes absolute, depending on where the camera is stored ).

Reply:
Our renderer abstraction actually has a "frame" option, so  the "frame + relativeShutter" idea could work (ironically I think we have "frame" because 3delight uses it implicitly as an AA seed, and then in the Arnold backend we copy "frame" into the explicit "AA_seed"). I'm not really sure we buy much by doing this transformation separately in each of the renderer backends rather than once in the translator though. And we have a separate debate about whether shutter should be in seconds so that the OSL time is in seconds, which muddies things further (do we really want to pass fps as well?). This isn't something that needs settling now, but I think perhaps the comment makes thing sound more settled than they are...